torch_geometric/nn/dense/dense_dmon_pool.py

```diff
@@ -0,0 +1,150 @@
+import torch
+from torch.nn import SELU, Dropout, Sequential
+
+from torch_geometric.nn.conv import MessagePassing
+from torch_geometric.nn.dense.linear import Linear
+
+EPS = 1e-15
+
+
+class DMonPooling(MessagePassing):
+    r"""Spectral modularity pooling operator from the `"Graph Clustering with
+    Graph Neural Networks" <https://arxiv.org/abs/2006.16904>`_ paper
+
+    .. math::
+        \mathbf{X}^{\prime} &= {\mathrm{softmax}(\mathbf{S})}^{\top} \cdot
+        \mathbf{X}
+
+        \mathbf{A}^{\prime} &= {\mathrm{softmax}(\mathbf{S})}^{\top} \cdot
+        \mathbf{A} \cdot \mathrm{softmax}(\mathbf{S})
+
+    based on dense learned assignments :math:`\mathbf{S} \in \mathbb{R}^{B
+    \times N \times C}`.
+    Returns learned cluster assignment matrix, pooled node feature matrix,
+    coarsened symmetrically normalized adjacency matrix, and two auxiliary
+    objectives: (1) The spectral loss
+
+    .. math::
+        \mathcal{L}_s = - \frac{1}{2m}
+        \cdot{\mathrm{Tr}(\mathbf{S}^{\top} \mathbf{B} \mathbf{S})}
+
+    where :math:`\mathbf{B}` is the modularity matrix, (2) the orthogonality
+    loss
+
+    .. math::
+        \mathcal{L}_o = {\left\| \frac{\mathbf{S}^{\top} \mathbf{S}}
+        {{\|\mathbf{S}^{\top} \mathbf{S}\|}_F} -\frac{\mathbf{I}_C}{\sqrt{C}}
+        \right\|}_F
+
+    where :math:`C` is the number of clusters, and (3) the cluster loss
+
+    .. math::
+        \mathcal{L}_c = \frac{\sqrt{C}}{n}
+        {\left\|\sum_i\mathbf{C_i}^{\top}\right\|}_F - 1.
+
+    Args:
+        dense_size (list): A 2D list whose each 1D list contains size for each
+            Linear layer being used.
+        k (int, optional): Number of clusters.
+            (default: :obj:`16`)
+        p (float, optional): Probability value for dropout layer.
+            (default: :obj:`0.2`)
+    """
+    def __init__(self, dense_size, k=16, p=0.2):
```

Comment:
Type hints :)

Reply:
Can you please help me with this just by giving a quick example of how to type hints for documentation as I don't see an example of it anywhere??

Reply:
I just mean
```python
def __init__(self, dense_size: int, k: int = 16, p: float = 0.2):
```